### PR TITLE
TT2-2023 Create a KMS key for an ECR repository

### PR DESCRIPTION
**Pull request by nacnudus**
Ticket Number: #[TT2-2023] 🎫

Following https://govukverify.atlassian.net/wiki/spaces/PLAT/pages/3107258369/How+to+deploy+a+container+to+Fargate+with+secure+pipelines#Step-3%3A-Create-a-repository-in-ECR

We need an ECR repository for an image that will be run as a task on Fargate. This key is to encrypt the image.

[TT2-2023]: https://govukverify.atlassian.net/browse/TT2-2023?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by richa-misra-gds — APPROVED** [review]

File: infrastructure/audit/template.yaml [hunk 1]
```diff
@@ -171,6 +171,38 @@ Resources:
       AliasName: !Sub alias/${AWS::StackName}/${Environment}/event-replay-sns-kms-key
       TargetKeyId: !Ref EventReplaySnsKmsKey
 
+  ReplayFromAthenaEcrKmsKey:
```
Comment (zaydgds) [inline infrastructure/audit/template.yaml:174]: most kms keys are stored in https://github.com/govuk-one-login/txma-infra/blob/main/infrastructure/core/template.yaml. if we store it here, it’ll put the keys in all our aws accounts so if future, if we need to use containers in other accounts, the keys will already be there. it'll also save us from having to migrate it over later

**Review by zaydgds — CHANGES_REQUESTED** [review]

File: infrastructure/core/template.yaml [hunk 2]
```diff
@@ -397,6 +397,38 @@ Resources:
...
+            Principal:
+              Service: ecr.amazonaws.com
+            Action:
+              - kms:Decrypt
+              - kms:Encrypt
+              - kms:GenerateDataKey*
+              - kms:DescribeKey
+            Resource: '*'
+
+  ReplayFromAthenaEcrKmsKeyAlias:
```
Comment (zaydgds) [inline infrastructure/core/template.yaml:426]: the key alias and resource names should probably be generic and follow a similar naming convention to the rest of the keys in this file

File: infrastructure/audit/template.yaml [hunk 3]
```diff
@@ -406,6 +406,13 @@ Resources:
       Type: String
       Value: !GetAtt AuditFileGlacierRestoreQueue.Arn
 
+  ReplayFromAthenaEcrKmsKeyArnParameter:
```
Comment (zaydgds) [inline infrastructure/audit/template.yaml:409]: missed a resource

**Review by zaydgds — CHANGES_REQUESTED** [review]

File: infrastructure/core/template.yaml (hunk 2 above)
Comment (zaydgds) [reply infrastructure/core/template.yaml:426]: > According to https://govukverify.atlassian.net/wiki/spaces/PLAT/pages/3107258369/How+to+deploy+a+container+to+Fargate+with+secure+pipelines#Step-3%3A-Create-a-repository-in-ECR we should have a separate repository per image, so I included the name of the repository/image in the name of the kay - @nacnudus 

ignore this

File: infrastructure/audit/template.yaml (hunk 3 above)
Comment (zaydgds) [reply infrastructure/audit/template.yaml]: > According to https://govukverify.atlassian.net/wiki/spaces/PLAT/pages/3107258369/How+to+deploy+a+container+to+Fargate+with+secure+pipelines#Step-3%3A-Create-a-repository-in-ECR we should have a separate repository per image, so I included the name of the repository/image in the name of the kay - @nacnudus 

ignore this comment

File: infrastructure/audit/template.yaml (hunk 1 above)
Comment (zaydgds) [reply infrastructure/audit/template.yaml]: > According to https://govukverify.atlassian.net/wiki/spaces/PLAT/pages/3107258369/How+to+deploy+a+container+to+Fargate+with+secure+pipelines#Step-3%3A-Create-a-repository-in-ECR we should have a separate repository per image, so I included the name of the repository/image in the name of the kay - @nacnudus 

ignore this comment

**Review by zaydgds — APPROVED** [review]

**Review by richa-misra-gds — APPROVED** [review]

**Pull request merged**